File: src/main/java/cz/wa/secretagent/world/entity/explosion/Explosion.java
package cz.wa.secretagent.world.entity.explosion;

import org.apache.commons.math3.geometry.euclidean.twod.Vector2D;

import cz.wa.secretagent.world.ObjectModel;
import cz.wa.secretagent.world.entity.Entity;
import cz.wa.secretagent.world.entity.EntityType;
import cz.wa.secretagent.world.entity.EntityType2;

/**
 * Explosion, has no second type, there is only one explosion.
 * 
 * @author Ondrej Milenovsky
 */
public class Explosion extends Entity {

    private final double radius;
    private final double damage;
    private double timeS;
    private final double durationS;

    public Explosion(ObjectModel model, Vector2D pos, double radius, double damage, double durationS) {
        super(model, pos);
        this.radius = radius;
        this.damage = damage;
        this.durationS = durationS;
        timeS = 0;
    }

    public double getRadius() {
        return radius;
    }

    public double getDamage() {
        return damage;
    }

    public void addTime(double timeS) {
        this.timeS += timeS;
    }

    public double getTimeS() {
        return timeS;
    }

    public double getDurationS() {
        return durationS;
    }

    @Override
    public EntityType getEntityType() {
        return EntityType.EXPLOSION;
    }

    @Override
    public EntityType2 getSecondType() {
        return null;
    }

}

File: src/main/kotlin/secretAgent/world/entity/Explosion.kt
package secretAgent.world.entity

import org.apache.commons.math3.geometry.euclidean.twod.Vector2D
import secretAgent.world.ObjectModel

/**
 * Explosion, has no second type, there is only one explosion.
 *
 * @author Ondrej Milenovsky
 */
class Explosion(model: ObjectModel, pos: Vector2D,
                val radius: Double,
                val damage: Double,
                val durationS: Double) : Entity(model, pos) {

    var timeS = 0.0
        private set

    override val type: EntityType
        get() = EntityType.EXPLOSION

    override val secondType: EntityType2?
        get() = null

    fun addTime(timeS: Double) {
        this.timeS += timeS
    }
}